scalalib/src/TestModule.scala

```diff
@@ -214,6 +214,17 @@ object TestModule {
     }
   }
 
+  /**
+   * TestModule that uses JUnit 5 Framework to run tests.
+   * You may want to provide the junit dependency explicitly to use another version.
+   */
+  trait Junit5 extends TestModule {
+    override def testFramework: T[String] = "net.aichler.jupiter.api.JupiterFramework"
+    override def ivyDeps: T[Agg[Dep]] = T {
+      super.ivyDeps() ++ Agg(ivy"net.aichler:jupiter-interface:0.9.0")
```

Comment:
At least it makes most user setups working without specifying this dependency, which probably makes them not aware of it which may lead to outdated tool chains... In other Scala test frameworks we don't specify the dependencies, but in Java test frameworks, we provide them, as we already bring the external sbt.testing implmentations, which transitively depend on the test runners.

Reply:
Can be a good think to create a separate target to make this configurable?
```scala
    def jupiterInterfaceIvyDep: T[Dep] = ivy"net.aichler:jupiter-interface:0.9.0"
    override def ivyDeps: T[Agg[Dep]] = T {
      super.ivyDeps() ++ Agg(jupiterInterfaceIvyDep())
    }
```

So if there is a new version out or they change the group-id people don't need to update Mill for that.
Not sure it's a good idea tho.

Reply:
Why not? It's the same as for Junit4 module.

Reply:
In general I don't like defaults which tie versions which have different lifecycles. It's always either a maintenance nightmare or the user has to provide more recent ones anyway.

Reply:
Am I right, that jupiter-interface will change less frequently that the version of JUnit 5? If yes, we can go the route @lolgab suggested, but I'd make it more flexible by using an `Agg[Dep]`.

```scala
def testFrameworkIvyDeps: T[Agg[Dep]] = T {
  Agg(ivy"net.aichler:jupiter-interface:0.9.0")
}
override def ivyDeps: T[Agg[Dep]] = T {
  super.ivyDeps() ++ testFrameworkIvyDeps()
}
```

If this will change rather often, we better don't include any default version!

Reply:
`testFrameworkIvyDeps` is too general to be defined here. May be defined in `TestModule` but I don't think it's a good idea either. Maybe if jupiter-interface doesn't change that often hardcoding isn't that bad after all.. 🙈

Reply:
yeah, was a too fast shot. `def jupiterInterfaceIvyDeps: T[Agg[Dep]]` then. This gives the option to make it empty or add more than one, if ever needed.